services/brig/src/Brig/Data/User.hs

```diff
@@ -404,9 +406,18 @@ lookupRichInfoMultiUsers users = do
 -- successful login.
 lookupUserTeam :: UserId -> AppIO (Maybe TeamId)
 lookupUserTeam u =
-  join . fmap runIdentity
+  (runIdentity =<<)
     <$> retry x1 (query1 teamSelect (params LocalQuorum (Identity u)))
 
+lookupUserUnverifiedEmail :: MonadClient m => UserId -> m (Maybe Email)
+lookupUserUnverifiedEmail userId =
+  (runIdentity =<<) <$> retry x1 (query1 cql (params LocalQuorum (Identity $ toUUID userId)))
+  where
+    cql :: PrepQuery R (Identity UUID) (Identity (Maybe Email))
+    cql =
+      "SELECT email \
+      \FROM vcodes WHERE account = ?"
```

Comment:
ah.  problem.  we can't use `account` in the where clause because it's not a key.  two solutions:

- add another database table
- make clients send the email after all, but look it up in vcode and assert that it matches on the right userid.

i strongly prefer the second.  easier on server load, easier to implement, more consistent with the other validation end-points no real downsides.

sorry, should have seen that earlier.  :(